libobs/callback/proc.c

```diff
@@ -79,14 +116,13 @@ bool proc_handler_call(proc_handler_t *handler, const char *name,
 	if (!handler)
 		return false;
 
-	for (size_t i = 0; i < handler->procs.num; i++) {
-		struct proc_info *info = handler->procs.array + i;
+	pthread_mutex_lock(&handler->mutex);
+	struct proc_info *info = getproc(handler, name);
+	pthread_mutex_unlock(&handler->mutex);
 
-		if (strcmp(info->func.name, name) == 0) {
-			info->callback(info->data, params);
-			return true;
-		}
-	}
+	if (!info)
+		return false;
 
-	return false;
+	info->callback(info->data, params);
```

Comment:
Have you considered covering this by the mutex too? Depending on where this is called from it may be possible another thread destroys the proc handler mid call, eg during shutdown.

Reply:
Yes it was actually something that I wasn't 100% sure should be done or not. I asked some users in the Discord, who said that they would personally prefer to use their own mutexes when necessary, but I'm open to other ideas.

The thread safety of signals and procs is not documented in any way. I think that the best idea may be to document what is threadsafe and what is not.